tests/testsuite/build.rs

```diff
@@ -2864,33 +2864,6 @@ fn recompile_space_in_name() {
     foo.cargo("build").with_stdout("").run();
 }
 
-#[cfg(unix)]
-#[cargo_test]
-fn credentials_is_unreadable() {
-    use cargo_test_support::paths::home;
-    use std::os::unix::prelude::*;
-    let p = project()
-        .file("Cargo.toml", &basic_manifest("foo", "0.1.0"))
-        .file("src/lib.rs", "")
-        .build();
-
-    let credentials = home().join(".cargo/credentials");
-    t!(fs::create_dir_all(credentials.parent().unwrap()));
-    t!(fs::write(
-        &credentials,
-        r#"
-            [registry]
-            token = "api-token"
-        "#
-    ));
-    let stat = fs::metadata(credentials.as_path()).unwrap();
-    let mut perms = stat.permissions();
-    perms.set_mode(0o000);
-    fs::set_permissions(credentials, perms).unwrap();
-
-    p.cargo("build").run();
```

Comment:
Why is this removed?

Reply:
With this change, `cargo` always reads the `credentials` file when it loads the other configuration.

I'm not sure what the intent behind supporting an unreadable credentials file is?

Reply:
I honestly don't remember either, and I suspect we're comfortable making this "breaking" change.
I will try and remember to bring it up at a Cargo Meeting, next time there's any room on the agenda.

Reply:
For reference PR #7774 and Issue #7624